pkg/util/k8s/k8s_test.go

```diff
@@ -901,6 +902,38 @@ func TestDeleteStatefulSet(t *testing.T) {
 	require.True(t, errors.IsNotFound(err))
 }
 
+func TestUpdateStorageClusterStatus(t *testing.T) {
+	k8sClient := testutil.FakeK8sClient()
+	cluster := &corev1alpha1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "test",
+			Namespace:       "test-ns",
+			ResourceVersion: "100",
+		},
+		Spec: corev1alpha1.StorageClusterSpec{
+			Image: "foo",
+		},
+	}
+
+	// Fail if cluster is not present
+	err := UpdateStorageClusterStatus(k8sClient, cluster)
+	require.True(t, errors.IsNotFound(err))
+
+	err = k8sClient.Create(context.TODO(), cluster)
+	require.NoError(t, err)
+
+	// Should update cluster if present with same resource version
```

Comment:
UpdateStorageClusterStatus is not checking for the same resource version, but simply overwriting the input with the existing resource version. 

May be you can add a test that actually updates the Status field of the StorageCluster and check if it got updated.

Reply:
Fixed the test to send input object with old resource version and verify that it gets updated to whatever is the latest in the api server.